api/v1alpha1/stack_type.go

```diff
@@ -4,122 +4,125 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+// +kubebuilder:resource:path=stack,scope=Namespaced,categories=stack,singular=stack
 // +kubebuilder:storageversion
 
-// Tinkerbell represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
+// Stack represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
 // Tinkerbell operator watches for different resources such as deployment, services, serviceAccounts, etc. One of those
-// CRs is Tinkerbell which the operator will install the tink-stack based on its specs. Once the CR is deleted,
+// CRs is Stack which the operator will install the tink-stack based on its specs. Once the CR is deleted,
 // the operator will delete all tinkerbell resources.
-type Tinkerbell struct {
+type Stack struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// Spec describes the desired tinkerbell stack state.
-	Spec TinkerbellStackSpec `json:"spec,omitempty"`
+	Spec StackSpec `json:"spec"`
 }
 
-// TinkerbellStackSpec specifies details of an Tinkerbell setup.
-type TinkerbellStackSpec struct {
+// StackSpec specifies details of an Tinkerbell setup.
+type StackSpec struct {
 	// Version is the Tinkerbell CRD version.
 	Version string `json:"version"`
 
 	// DNSResolverIP is indicative of the resolver IP utilized for setting up the nginx server responsible for proxying
 	// to the Tinkerbell service and serving the Hook artifacts.
-	DNSResolverIP string `json:"dnsResolverIP"`
+	DNSResolverIP string `json:"dnsResolverIP,omitempty"`
 
 	// Registry is the registry to use for all images. If this field is set, all tink service deployment images
 	// will be prefixed with this value. For example if the value here was set to docker.io, then boots image will be
 	// docker.io/tinkerbell/boots.
-	Registry string `json:"registry"`
+	Registry string `json:"registry,omitempty"`
 
-	// ImagePullSecret the secret name containing the docker auth config which should exist in the same namespace where
+	// ImagePullSecrets the secret name containing the docker auth config which should exist in the same namespace where
 	// the operator is deployed(typically tinkerbell)
-	ImagePullSecret []string `json:"imagePullSecret"`
+	ImagePullSecrets []string `json:"imagePullSecrets,omitempty"`
 
 	// Services contains all Tinkerbell Stack services.
-	Services Services `json:"services"`
+	Services Services `json:"services,omitempty"`
 }
 
 // Services contains all Tinkerbell Stack services.
 type Services struct {
 	// Boots contains all the information and spec about boots.
-	Boots Boots `json:"boots"`
+	Boots Boots `json:"boots,omitempty"`
 
 	// Hegel contains all the information and spec about boots.
-	Hegel Hegel `json:"hegel"`
+	Hegel Hegel `json:"hegel,omitempty"`
 
 	// Rufio contains all the information and spec about rufio.
-	Rufio Rufio `json:"rufio"`
+	Rufio Rufio `json:"rufio,omitempty"`
```

Comment:
I think the only 2 components that are actually required are Tink Server and Tink Controller. They provide the bare minimum deployment that brings value. Normally Boots will be deployed but it is optional.

Reply:
Not quite sure how this is relevant to this line here though. As you can see, the field has `omitempty` which means it is optional.

Reply:
In addition, the two fields, `TinkServer` and `TinkController` only have an image and tls enabling which can be defaulted without the need to assign any values by the users. The only thing that can't accept defaulting in my opinion, it should be the version, as it is the field which decide the stack components versions to be deployed based on matrix that we need to document.